src/Symfony/Component/DependencyInjection/ParameterBag/ContainerBag.php

```diff
@@ -40,7 +42,11 @@ public function all()
      */
     public function get(string $name)
     {
-        return $this->container->getParameter($name);
+        try {
+            return $this->container->getParameter($name);
```

Comment:
I think `Container::getParameter` should be the one for which the docblock must be fixed, to indicates it throws a `ParameterNotFoundException` rather than a simple `InvalidArgumentException`.

Reply:
Yes, i agree too. I'll update this.
Maybe directly update `ContainerInterface` too?

Reply:
It maybe more complex than its seems (ive missed something at first). I need your opinion.

The basic `Container` throw the right exception `ParameterNotFoundException` as excepted, it is just the phpdoc that is not right.

But, there really is one container class that throw `InvalidArgumentException` instead of `ParameterNotFoundException`, it is the one generated by the `PhpDumper`, look at [PhpDumper#1531](https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php#L1531=), [PhpDumper#1580](https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php#L1580=) and [PhpDumper#1539](https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php#L1539=).

I have update the PR to really fixthe throw this time. I have updated the fixtures files for dumper and tests passed.

Is this ok?

Reply:
These changes look right to me. We're loosing the "dynamic parameter" info from the original exception message, but I'm not convinced it matters much.

Reply:
To prevent loosing the info of dynamic parameter, i can create a new exception that extend `ParameterNotFoundException`, something like `MissingDynamicParameterException`.